src/vcf/assembly_checker.cpp

```diff
@@ -99,9 +122,58 @@ namespace ebi
           return is_valid;
       }
 
+      std::vector<std::string> get_matching_synonyms_list(ebi::assembly_report::Synonyms_map &synonyms_map,
+                                  size_t line_num,
+                                  RecordCore &record_core,
+                                  bioio::FastaIndex &fasta_index,
+                                  std::vector<std::unique_ptr<ebi::vcf::AssemblyCheckReportWriter>> &outputs)
+      {
+          std::vector<std::string> found_synonyms;
+
+          bool is_contig_available = synonyms_map.is_contig_available(record_core.chromosome);
+          if (!is_contig_available) {
+              found_synonyms.push_back(record_core.chromosome);
+              return found_synonyms;
```

Comment:
@cyenyxe do you think it would be useful to raise this as an error and stop the program? I could see that the assembly report provided is from the wrong species, and all contig synonyms are silently unavailable, and it could fail later as "contig missing from FASTA" or other misleading error message

Reply:
>  a name can't be found in the assembly report at all
exactly